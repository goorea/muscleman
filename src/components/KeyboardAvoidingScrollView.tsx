import React from 'react';
import { KeyboardAvoidingView, Platform, ScrollView } from 'react-native';
import { HeaderHeightContext } from '@react-navigation/elements';

const KeyboardAvoidingScrollView: React.FC = ({ children }) => {
  const headerHeight = React.useContext(HeaderHeightContext);

  return (
    <KeyboardAvoidingView
      behavior={Platform.OS === 'ios' ? 'padding' : undefined}
      keyboardVerticalOffset={Platform.OS === 'ios' ? headerHeight : 0}>
      <ScrollView>{children}</ScrollView>
    </KeyboardAvoidingView>
  );
};

export default KeyboardAvoidingScrollView;
