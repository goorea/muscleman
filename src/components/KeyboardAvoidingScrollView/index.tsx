import React from 'react';
import { Platform, ScrollView } from 'react-native';
import { HeaderHeightContext } from '@react-navigation/elements';
import { Container } from './styled';

type P = {
  floatChildren?: React.ReactNode;
};

const KeyboardAvoidingScrollView: React.FC<P> = ({
  children,
  floatChildren,
}) => {
  const headerHeight = React.useContext(HeaderHeightContext);

  return (
    <Container
      behavior={Platform.OS === 'ios' ? 'padding' : undefined}
      keyboardVerticalOffset={Platform.OS === 'ios' ? headerHeight : 0}>
      <ScrollView>{children}</ScrollView>
      {floatChildren}
    </Container>
  );
};

export default KeyboardAvoidingScrollView;
