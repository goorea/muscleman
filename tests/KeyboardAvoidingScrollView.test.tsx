import 'react-native';
import React from 'react';
import { render } from '@testing-library/react-native';
import KeyboardAvoidingScrollView from '@src/components/KeyboardAvoidingScrollView';

describe('KeyboardAvoidingScrollView 컴포넌트', () => {
  it('렌더링이 올바르게 된다', () => {
    const { toJSON } = render(<KeyboardAvoidingScrollView />);

    expect(toJSON()).toMatchSnapshot();
  });
});
