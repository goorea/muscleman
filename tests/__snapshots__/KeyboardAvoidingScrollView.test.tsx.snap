// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`KeyboardAvoidingScrollView 컴포넌트 렌더링이 올바르게 된다 1`] = `
<View
  onLayout={[Function]}
  style={
    Object {
      "paddingBottom": 0,
    }
  }
>
  <RCTScrollView>
    <View />
  </RCTScrollView>
</View>
`;
